generator/ServiceClientGeneratorLib/Shape.cs

```diff
@@ -188,7 +188,40 @@ public Shape KeyShape
                 return this.model.FindShape(extendsNode.ToString());
             }
         }
-
+        /// <summary>
+        /// Gets the map's key node's xmlnamespace.
+        /// </summary>
+        public string KeyShapeXmlNamespace
+        {
+            /*
+            "type": "map",
+            "key": {
+                "shape": "String",
+                "locationName": "K",
+                "xmlNamespace": "https://the-key.example.com"
+            },
+             */
+            get
+            {
+                var keyNode = this.data[KeyKey];
+                if (keyNode == null || keyNode[ServiceModel.XmlNamespaceKey] == null)
+                    return "";
+                return (string)keyNode[ServiceModel.XmlNamespaceKey];
+            }
+        }
+        /// <summary>
+        /// Gets the map's value node's xmlnamespace
+        /// </summary>
+        public string ValueShapeXmlNamespace
+        {
+            get
+            {
+                var valueNode = this.data[ValueKey];
+                if (valueNode == null || valueNode[ServiceModel.XmlNamespaceKey] == null)
+                    return "";
+                return (string)valueNode[ServiceModel.XmlNamespaceKey];
+            }
+        }
```

Comment:
These don't check for IsMap. Should the base methods be modified instead with specific instructions for `IsMap` type shapes/members"?

Reply:
hmm I think that might be better actually, let me try to make that change and see if it works

Reply:
Spoke to Bo offline and we agreed that since this is only valid on a shape, it shouldn't be on the base class. Instead I just added a check for ISMap